### PR TITLE
include Kotlin DSL sample for chapter build_lifecycle

### PR DESCRIPTION
**Pull request by jnizet**
Note: I added a TODO (and should have done it too on the last commit) because two groovy closures take 2 arguments (the second one being the state of the project/task), and that can't be done the same way in Kotlin. Fortunately, the state is available as a property of the Project/Task, so that can be translated the same way.

I'm also not sure if the way I chose to set/get the properties of projects and tasks is the best way and has the exact same semantics as in the groovy sample. 

### Context
refs #6442 

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/build_lifecycle.adoc [hunk 1]
```diff
@@ -199,15 +169,13 @@ This example uses method `Project.afterEvaluate()` to add a closure which is exe
...
-include::{samplesPath}/userguide/buildlifecycle/buildProjectEvaluateEvents/build.gradle[tag=evaluate-events]
-----
+.Notifications
+====
+include::sample[dir="userguide/buildlifecycle/buildProjectEvaluateEvents/groovy",files="build.gradle[tags=evaluate-events]"]
+include::sample[dir="userguide/buildlifecycle/buildProjectEvaluateEvents/kotlin",files="build.gradle.kts[tags=evaluate-events]"]
+====
 
-.Output of **`gradle -q test`**
+.Output of **`gradle -q test`** (partial)
```
Comment (eskatos) [inline subprojects/docs/src/docs/userguide/build_lifecycle.adoc]: :x: why not display the full outputs as it was?

If it's about the different file names in the output between groovy/kotlin, then you should have two output files.

File: subprojects/docs/src/samples/userguide/buildlifecycle/buildProjectEvaluateEvents/kotlin/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// TODO not exactly the same as groovy: how to handle the second argument in a type-safe way?
```
Comment (eskatos) [inline subprojects/docs/src/samples/userguide/buildlifecycle/buildProjectEvaluateEvents/kotlin/build.gradle.kts]: You did it right. I'd modify the Groovy sample to match the Kotlin one here:

```groovy
// groovy
// tag::evaluate-events[]
gradle.afterProject { project ->
    if (project.state.failure) {
        println "Evaluation of $project FAILED"
    } else {
        println "Evaluation of $project succeeded"
    }
}
// end::evaluate-events[]

task test
```

File: subprojects/docs/src/samples/userguide/buildlifecycle/projectEvaluateEvents/kotlin/projectA.gradle.kts [hunk 3]
```diff
@@ -0,0 +1 @@
+setProperty("hasTests", true)
```
Comment (eskatos) [inline subprojects/docs/src/samples/userguide/buildlifecycle/projectEvaluateEvents/kotlin/projectA.gradle.kts]: :x: should be `extra["hasTests"] = true` to match the root project using extra properties explicitely

File: subprojects/docs/src/samples/userguide/buildlifecycle/taskCreationEvents/kotlin/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+tasks.whenTaskAdded {
+    extra["srcDir"] = "src/main/java"
+}
+
+val a = task("a")
+
+println("source dir is ${a.property("srcDir")}")
```
Comment (eskatos) [inline subprojects/docs/src/samples/userguide/buildlifecycle/taskCreationEvents/kotlin/build.gradle.kts]: :x: should be `a.extra["srcDir"]`

File: subprojects/docs/src/docs/userguide/build_lifecycle.adoc (hunk 1 above)
Comment (eskatos) [reply subprojects/docs/src/docs/userguide/build_lifecycle.adoc]: The examplar tests do normalize absolute paths from the output:

https://github.com/gradle/gradle/blob/bc3f89dc629f64ec37b682341d3d741bfe0cc642/subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/samples/UserGuideSamplesIntegrationTest.groovy#L32-L41

and

https://github.com/gradle/gradle/blob/bc3f89dc629f64ec37b682341d3d741bfe0cc642/subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/logging/SampleOutputNormalizer.groovy#L26-L27

Comment (jnizet) [reply subprojects/docs/src/docs/userguide/build_lifecycle.adoc]: You're right, of course. And my diagnostic was wrong. Fixed now.

Comment (eskatos) [reply subprojects/docs/src/docs/userguide/build_lifecycle.adoc]: Great, thanks!
The added ` (partial)` on samples titles should then be removed too.
They were other samples outputs truncated in this PR, could you fix them all?

Comment (jnizet) [reply subprojects/docs/src/docs/userguide/build_lifecycle.adoc]: Oops, sorry for missing that. Done now.

File: subprojects/docs/src/samples/userguide/buildlifecycle/projectEvaluateEvents/kotlin/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+// tag::after-evaluate[]
+allprojects {
+    afterEvaluate {
+        if (property("hasTests") as Boolean) {
```
Comment (eskatos) [inline subprojects/docs/src/samples/userguide/buildlifecycle/projectEvaluateEvents/kotlin/build.gradle.kts]: :x: should be `extra["hasTests"]`, sorry for missing that one in my previous review

Comment (jnizet) [reply subprojects/docs/src/samples/userguide/buildlifecycle/projectEvaluateEvents/kotlin/build.gradle.kts]: Done @eskatos

**Pull request merged**